### ISSUE TITLE
Was there a significant change in react-addons-perf function getInclusive between React 15.0.2 and 15.4.2?

### ISSUE THREAD

**Issue by ender2012** [facebook/react#8867]
_Note: I asked this on SO first, but was directed here._

TL;DR: Did the massive overhaul of react-addons-perf make a large change that would cause wildly different results on the same data using Perf.getInclusive?

We are in the process of upgrading from React 15.0.2 to React 15.4.2 and we are seeing some puzzling changes when profiling using react-perf-addons. Our test case is a table with 200 rows and 2200 cells.

After upgrading we tested total render times using Window.performance and using react-addons-perf’s getLastMeasurment.duration. Both of these tools show a 10% – 15% increase in performance when rendering the table; this does not seem outside of the realm of possibility.

Things started to get weird when we looked at the results from Perf.printInclusive. The results from printInclusive (for each individual component render) report that they render faster by a factor of 5 and this does not seem to jive with what we are seeing elsewhere.

I tried to look at the changes inside the ReactPerf.js module but this file was replaced wholesale so it is hard to understand what may have changed and nothing in the changelog would really explain this big of difference in the results from printInclusive.

My question is, could these grossly different findings from Perf.printInclusive be explained by the big refactor on react-addons-perf? Is this function showing something different from what it used to show? Were the old numbers wrong?

**Comment by gaearon** [comment]
In 15.0 Perf was measuring both library code time and application code time.

Since 15.1 it is measuring just the application code time which is more actionable.
So it only includes time spent in your components, not in React code itself.

If you're interested in the full time, you can learn that information with [timeline integration](https://facebook.github.io/react/docs/optimizing-performance.html#profiling-components-with-chrome-timeline).

I hope this helps!

**Comment by ender2012** [comment]
👍  Thank you so much for the clarification, this was very helpful!

**Comment by limscoder** [comment]
@gaearon Makes sense, but I'm not sure I like the change, since total duration tends to be important for estimating if you can hit initial render and framerate guidelines, as well as comparing different versions of react. While it's not actionable in the sense of being able to optimize react's internals, it is actionable in the sense of "this update meets our performance guidelines" or "this update results in inefficient react dom update and we should probably do something else".

Is recording both numbers out of the question?

**Comment by gaearon** [comment]
If you just need to record initial render, why not use the browser APIs for this?

**Comment by ender2012** [comment]
@gaearon I think the issue is that we like to have sub-component level render times and you can get this out of Perf.printInclusive but it no longer gives you the total render time.

**Comment by gaearon** [comment]
Thanks for the feedback. I see how both kinds of information could be valuable. We likely won't be reverting the behavior at this point as we're in the middle of a core engine rewrite but we'll consider this feedback as we get to making ReactPerf work with it.
